### PR TITLE
Use BLE implementation in DeviceLayer in CHIP Device Controller Python Binding

### LINKED ISSUE #4399: for python device controller, replace ChipCoreBluetoothMgr.py with the same implementation as iOS app has.
Sorry by device I mean an iOS device. chip-tool simulates a controller, like the iOS device controller. And on iOS, src/platform/darwin is the used BLE stack.
-- Yep, agree, for python device controller, we should replace ChipCoreBluetoothMgr.py with the same implementation as iOS app has.

_Originally posted by @yunhanw-google in https://github.com/project-chip/connectedhomeip/issues/4334#issuecomment-761197450_

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
Sorry by device I mean an iOS device. chip-tool simulates a controller, like the iOS device controller. And on iOS, src/platform/darwin is the used BLE stack.
-- Yep, agree, for python device controller, we should replace ChipCoreBluetoothMgr.py with the same implementation as iOS app has.

Originally posted in #4334 (comment)

 #### Summary of Changes
Use BLE implementation in DeviceLayer in CHIP Device Controller Python Binding, this does not remove the BLEManager classes, they are used for commands like "ble-scan" now.

There are verbose output in device controller, and some cleanup is required as follow up.

This PR also does similar things with Linux.

 Fixes #4399

### CODE REVIEW

File: src/controller/python/chip/ChipCoreBluetoothMgr.py [hunk 1]
```diff
@@ -538,88 +520,16 @@ def bgScanStop(self):
...
-
-                    break
-
-        ret = True if self.loop_condition and self.connect_state else False
-
-        resString = "connect " + ("success" if ret else "fail")
-        self.logger.info(resString)
-
-        return ret
+        """ No longer supports ble connect in chip device controller """
```
Comment (andy31415) [inline src/controller/python/chip/ChipCoreBluetoothMgr.py:523]: Consider instead:

```py
""" DEPRECATED. Connect not supported and will be removed."""
```

and a GH TODO isse to remove (or explain what the plan is for the future) 

Same for the entries below.

if not supported, why are we not removing outright?

File: src/controller/python/chip/ChipDeviceCtrl.py [hunk 2]
```diff
@@ -39,14 +39,6 @@
 
 _CompleteFunct = CFUNCTYPE(None, c_void_p, c_void_p)
```
Comment (andy31415) [inline src/controller/python/chip/ChipDeviceCtrl.py:40]: Lets remove unused imports as well as part of this PR (time and Thread)

Also global variables (_CompleteFunct and _ErrorFunct according tothe code scanner)

File: src/transport/BLE.cpp [hunk 3]
```diff
@@ -164,6 +165,9 @@ void BLE::OnBleConnectionComplete(void * appState, BLE_CONNECTION_OBJECT connObj
     CHIP_ERROR err = CHIP_NO_ERROR;
     BLE * ble      = reinterpret_cast<BLE *>(appState);
 
+    // We have already setup the ble endpoint for the same object, this should not happen, but we can tolerate it.
+    VerifyOrReturn(ble->mBleEndPoint == nullptr || !ble->mBleEndPoint->ConnectionObjectIs(connObj));
```
Comment (andy31415) [inline src/transport/BLE.cpp:169]: Could we have an explanation on when this happens?

If it should never happen, we should probably print out a warning. This seems to silently ignore a state of 'this should not happen'

File: src/controller/python/chip/ChipCoreBluetoothMgr.py (hunk 1 above)
Comment (erjiaqing) [reply src/controller/python/chip/ChipCoreBluetoothMgr.py]: Done, and created 4548 for deeper clean up.

File: src/transport/BLE.cpp (hunk 3 above)
Comment (erjiaqing) [reply src/transport/BLE.cpp]: Done, also created 4547 for final fix.

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by woody-apple** [comment]
@erjiaqing Can you take a peek at the conflict here please? 
@jelderton @saurabhst can you take a peek man?

**Review by saurabhst — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**